source/usr/lib/mcvirt/node/network/factory.py

```diff
@@ -66,11 +67,21 @@ def create(self, name, physical_interface):
         if self.check_exists(name):
             raise NetworkAlreadyExistsException('Network already exists: %s' % name)
 
-        # Ensure that the physical interface eixsts
+        if self._is_cluster_master:
+            def remote_command(remote_connection):
+                network_factory = remote_connection.get_connection('network_factory')
+                cluster = remote_connection.get_connection('cluster')
+                if not network_factory.interface_exists(physical_interface):
+                    raise InterfaceDoesNotExist(
+                        'Physical interface %s does not exist on remote node: %s'
+                        % (physical_interface, cluster.generate_connection_info()[0]))
```

Comment:
Please don't run cluster.generate_connection_info() to get the hostname. Probably best to create a method called, assert_exists, which runs the _interface_exists method and raises the exception if it doens't exist, then you just need to run this on the remote ndoe